src/Microsoft.AspNet.Mvc.TagHelpers/Internal/FileVersionProvider.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Security.Cryptography;
+using Microsoft.AspNet.FileProviders;
+using Microsoft.AspNet.WebUtilities;
+using Microsoft.Framework.Caching.Memory;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers.Internal
+{
+    /// <summary>
+    /// Provides version hash for a specified file.
+    /// </summary>
+    public class FileVersionProvider
+    {
+        private const string VersionKey = "v";
+        private readonly IFileProvider _fileProvider;
+        private readonly string _applicationName;
+        private readonly IMemoryCache _cache;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="FileVersionProvider"/>.
+        /// </summary>
+        /// <param name="fileProvider">The file provider to get and watch files.</param>
+        /// <param name="applicationName">Name of the applicaiton.</param>
+        /// <param name="cache">Cache where versioned urls of files are cached.</param>
+        public FileVersionProvider(
+            [NotNull] IFileProvider fileProvider,
+            [NotNull] string applicationName,
+            IMemoryCache cache)
```

Comment:
why is an instance of this class instantiated when caller does not have a `IMemoryCache` instance?  what is the scenario when file hashes should be generated but not cached?